Automattic/wp-calypso
# P2: new signup flow

#### Changes proposed in this Pull Request

* Partially implements the new signup flow as described in p9lV3a-2Lp-p2 :
    * Defines a new P2 flow (`p2-new`) that adds a new step and  puts user signup/login before site creation
    * Note: the old P2 flow (`p2`) will not _yet_ be removed, and will still be the main flow, until more steps are implemented for the new flow.

#### Testing instructions
`yarn install && ENABLE_FEATURES=no-force-sympathy yarn start` or Calypso live

1. Regression test: make sure that the old flow, `/start/p2`, still works as expected. Go to `calypso.localhost:3000/start/p2` and go through the signup flow. 
2. Testing the new flow:
    - Note: `/start/p2-new` is not the complete and final version of the revised flow described in p9lV3a-2Lp-p2. It will not yet be linked to by the P2 marketing page. However, it is a complete flow, i.e. it allows users to log in or sign up, and create P2 workspaces.
    - To test, go to `calypso.localhost:3000/start/p2-new` and go through the signup flow.





#### Screens
<img width="1674" alt="Screen Shot 2021-11-29 at 7 23 15 PM" src="https://user-images.githubusercontent.com/730823/143859539-f792ffac-dde3-4364-829c-c0331ed27c56.png">
